### DISCUSSION TITLE
خطا در متد RequestAsync

### DISCUSSION THREAD

**Discussion by smhkashanchi** [Sina-Soltani/Parbad#260] [Q&A]
سلام
من داشتم از کدهای شما برای درگاه ملی استفاده می کردم که ‍‍پس از انجام تنظیمات به خطای زیر می خورم.
لطفا راهنمایی بفرمیید
متن خطا:
MelliBank signing data failed. Specified key is not a valid size for this algorithm. (Parameter 'rgbKey') in parbad gateway

اینم کد من قبل از رفتن به درگاه بانک:
`            var callbackUrl = Url.Action("Verify", "Cart", null, Request.Scheme);

            var Bankresult = await _onlinePayment.RequestAsync(invoice =>
            {
                invoice
                    .SetAmount(factor.TotalPrice * 10)
                                    .SetCallbackUrl(callbackUrl)
                .SetTrackingNumber(Convert.ToInt64(factor.Factor_Id))
                .UseMelli();
            });`

با تشکر

**Comment by Sina-Soltani** [comment]
درود
TerminalKey 
رو برای این درگاه تنظیم کردید؟

نوع برنامه چیه؟
ASP.NET Core?

سیستم عامل؟

**Reply by smhkashanchi** [reply]
بله
کد استارتاپ:
` services.AddParbad()
                 .ConfigureGateways(gateways =>
                 {
                     gateways
                         .AddMelli()
                         .WithAccounts(accounts =>
                         {
                             accounts.AddInMemory(account =>
                             {
                                 account.Name = "Melli"; // optional if there is only 1 account for this gateway
                                 account.TerminalId = "*******";
                                 account.MerchantId = "***********";
                                 account.TerminalKey = "***********************************";
                             });
                         });

                     gateways
                         .AddParbadVirtual()
                         .WithOptions(options => options.GatewayPath = "/MyVirtualGateway");
                 })
                 .ConfigureHttpContext(builder => builder.UseDefaultAspNetCore())
                 .ConfigureStorage(builder => builder.UseMemoryCache());`

پروژه دات نت کور 3.0 هست و سیستم عامل هم ویندوز

**Reply by Sina-Soltani** [reply]
(لطفا اطلاعات خصوصی درگاه رو هیچوقت در مکان عمومی ننویسید)

بهتره با پشتیبانی این درگاه صحبت کنید و ببینید آیا کلید رمزنگاری صحیح به شما دادند یا خیر
مشابه این مشکل قبلا بوده و مشکل از کلید رمزنگاری بوده
نمونه:
https://github.com/Sina-Soltani/Parbad/issues/255

**Comment by smhkashanchi** [comment]
بله درست می فرمایید.
اصلاح کردم.

**Comment by smhkashanchi** [comment]
سلام مجدد
من هر کاری می کنم بازم خطای Access-Control-Allow-Origin میده و درخواست به سرور نمیره؟
تو نت هم سرچ زدم ولی نتیجه نداد.
به نظر شما مشکل از چیه ؟؟
با تشکر

**Comment by Sina-Soltani** [comment]
درود

این پیغام خطا رو
از کجا؟
از چه چیزی؟
و در چه مرحله ای دریافت میکنید؟

**Comment by smhkashanchi** [comment]
همون مرحله اول  قبل از اینکه بره درگاه در متد زیر:
` var Bankresult = await _onlinePayment.RequestAsync(invoice =>
                {
                    invoice
                        .SetAmount(factor.TotalPrice * 10)
                        //.SetAmount(50001)
                        .SetCallbackUrl(callbackUrl)
                    .SetTrackingNumber(Convert.ToInt64(factor.Factor_Id))
                    .UseMelli();
                });`

` if (Bankresult.IsSucceed)
                {
                    return Bankresult.GatewayTransporter.TransportToGateway();
                }`

هر چی سرچ کردم و تنظیمات استارت اپ و اکشن را هم انجام دادم بازم درست نمیشه.
نمی دونم مشکلش چیه

لطفا بررسی بفرمایید.
با تشکر

**Reply by Sina-Soltani** [reply]
لطفا یک اسکرین شات از جایی که پیغام
Access-Control-Allow-Origin
رو دریافت میکنید بگیرید و اینجا قرار بدید

**Reply by smhkashanchi** [reply]
![image](https://user-images.githubusercontent.com/93983325/141614936-486a1157-c722-42a1-b329-54b646515da7.png)
![image](https://user-images.githubusercontent.com/93983325/141614953-7e40500e-8df1-4fad-999e-badba9b18ca3.png)

**Reply by Sina-Soltani** [reply]
لطفا از خروجی متد
RequestAsync
که در مثال شما در شئ
Bankresult
قرار گرفته هم یک عکس بزارید

همچنین پراپرتی های این شي کاملا مشخص باشه

**Reply by smhkashanchi** [reply]
![image](https://user-images.githubusercontent.com/93983325/141615161-a0e03d92-3fa0-4725-addc-d381865e6052.png)
![image](https://user-images.githubusercontent.com/93983325/141615182-9e8475e9-c32f-424e-934a-247b700f5cf6.png)

اگر بخواهید با انی دسک نیز می توانید وصل بشید به سیستم من

**Comment by Sina-Soltani** [comment]
شما گفتید که پیغام خطا رو از متد و این پکیج دریافت میکنید
همونطور که در پراپرتی ها مشاهده میکنید هیچگونه خطایی وجود نداره و نتیجه پراپرتی IsSucceed هم موفق هست و هیچگونه پیغام خطایی هم وجود نداره
شما اون پیغام خطا رو از مرورگر دارید دریافت میکنید نه از برنامه یا این پکیج

بنابراین خطایی که شما در حال حاضر دارید در واقع هیچگونه ارتباطی با  این پکیج نداره و مربوط به برنامه شما هست
اگر پیغام خطا رو توی گوگل سرچ کنید میبینید که به شما گفته میشه که CORS رو در برنامه خودتون فعال کنید
https://docs.microsoft.com/en-us/aspnet/core/security/cors?view=aspnetcore-6.0
